### ISSUE TITLE
Suggested addons in settings page

### ISSUE THREAD

**Issue by lisa-wolfgang** [ScratchAddons/ScratchAddons#3444]
### Is your request related to a problem? Please describe.

There are so many addons that users may not have time to scroll through them all, and they may have a change of mind in the future about addons of that type or function.

### Describe the solution you'd like

At the very top of the settings page, there would be a "Suggested for you" section that would list addons that SA thinks the user would like.

We would need to discuss, of course, what these suggestions would be sourced from. Assume that these are all calculated when the settings page is loaded (based on a `seeAlso` or `related` field in each addon manifest) and that the suggestions don't dynamically update as the user toggles settings. There are a couple of options:

- **The user's addon configuration.** This would be the easiest and least invasive to implement, but it would only really work if the user actually went through the page and enabled addons, and it's less reflective of a user's current preferences, which defeats the entire purpose of this feature.
- **The user's recently enabled addons.** This would also be easy and noninvasive to implement -- [W_L is already in favor of having a log of recent settings changes for troubleshooting purposes](https://discord.com/channels/751206349614088204/751207339545198682/884093297746903060). This would more presently reflect what kinds of features a user is interested in, provided that new addons are relatively balanced in type. Not the most robust solution, but it comes without any real tradeoffs, which probably makes it the best option.
- **How much time the user spends on each type of Scratch page.** (i.e. studios, forums, code editor, costume editor) This would be relatively easy but somewhat invasive, likely requiring a change to the privacy policy. (If I'm guessing correctly, you read the words "privacy policy" and instantly rejected this option.) The benefit would be that the user wouldn't need to enable addons in order for suggestions to be made.
- **How much each addon is used.** This would probably be somewhat cumbersome to implement and is pretty invasive in general, and it doesn't even make sense for some types of addons, but the one benefit would be that addons with no usage could appear as suggestions to disable for performance / load time improvement.
- **Multiple of the above** to take advantage of each option's benefits.

### Additional context

This may not solve the problem of the settings' organization, but even the best organization method takes some degree of time to browse through, especially as our addon count continues to grow with no intention of combining features.

**Comment by apple502j** [comment]
"How much time the user spends on each type of Scratch page" and "How much each addon is used" are rejected for privacy concerns (especially on AMO).

**Comment by WorldLanguages** [comment]
I think we should first focus on something like this, but for first install.

**Comment by LankyBox01** [comment]
Another option is making an AI to study someone's behaviour on enabling/disabling addons, then make a small list of addons that the user might like, based on recently used ones.

**Comment by WorldLanguages** [comment]
> Another option is making an AI to study someone's behaviour on enabling/disabling addons, then make a small list of addons that the user might like, based on recently used ones.

Serious suggestions only please :P

**Comment by LankyBox01** [comment]
> > Another option is making an AI to study someone's behaviour on enabling/disabling addons, then make a small list of addons that the user might like, based on recently used ones.
> 
> Serious suggestions only please :P

Ok sorry

**Comment by Samq64** [comment]
I think adding
```json
{
  "seeAlso": ["addon-id"]
}
```
to some addons and listing them under the presets would be the best option because it's simple and wouldn't cause any privacy concerns. This would also mean notices such as "If you want to see more projects and curators at once, use the 'more items per row' addon." unnecessary because they could just go in the see also section.

**Comment by mxmou** [comment]
> I think adding
> 
> ```json
> {
>   "seeAlso": ["addon-id"]
> }
> ```
> 
> to some addons and listing them under the presets would be the best option because it's simple and wouldn't cause any privacy concerns. This would also mean notices such as "If you want to see more projects and curators at once, use the 'more items per row' addon." unnecessary because it could just go in the see also section.

I think being to be able to add a description to each addon listed would be useful to explain how it is related.

**Comment by CST1229** [comment]
> > I think adding
> > ```json
> > {
> >   "seeAlso": ["addon-id"]
> > }
> > ```
> > to some addons and listing them under the presets would be the best option because it's simple and wouldn't cause any privacy concerns. This would also mean notices such as "If you want to see more projects and curators at once, use the 'more items per row' addon." unnecessary because it could just go in the see also section.
> 
> I think being to be able to add a description to each addon listed would be useful to explain how it is related.

In that case, possible structure:
```json
{
	"seeAlso": [
		{
			"id": "items-per-row",
			"relation": "To see more projects and curators at once."
		},
		{
			"id": "dark-www",
			"relation": "To use colors similar to the Scratch 2.0 pages"
		}
	]
}
```

**Comment by Samq64** [comment]
That could work. `description` and `addon` could be used.
```json
{
  "seeAlso": [
    {
      "addon": "items-per-row",
      "description": "To see more projects and curators at once."
    },
    {
      "addon": "dark-www",
      "description": "To use colors similar to the Scratch 2.0 pages."
    }
  ]
}
 ```

**Comment by lisa-wolfgang** [comment]
`seeAlso` is a dependency of this feature, so perhaps it should be discussed in its own issue: #3933

**Comment by jasonbrown1965** [comment]
happy to open an issue 
but just double checking? 
to avoid and/all of my 
inevitable misunderstandings : 

issue: do any of these privacy issues also raise security issues ? 

#eli5 all that code stuff was way above my paygrade, sorry. 
happy to read links/sources, just please do 
plain lanuage intro/s?  
e.g. a step down from 'idiot' guides
more a 'drooling idiot' audience, 
such as myself 

or is there another forum that might be better to raise this in?

**Comment by DNin01** [comment]
I think it would be so awesome if you were doing something (like maybe browsing a project with a lot of code) and then Scratch Addons knew that you were doing that thing and then when you go into the settings page looking for the addon you want, you immediately see a suggestion for the thing you wanted (e.g. Find Bar) in the settings page, like magic.

**Comment by Samq64** [comment]
#6104 has been merged which is a very good start.
